var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -122,3 +123,19 @@ def cmake_args(self):
         ]
 
         return args
+
+    @property
+    def libs(self):
+        shared = '+shared' in self.spec
+        if 'dims=2' in self.spec:
+            return find_libraries(
+                ['libwarpx.2d'], root=self.prefix, recursive=True, shared=shared
+            )
+        if 'dims=3' in self.spec:
+            return find_libraries(
+                ['libwarpx.3d'], root=self.prefix, recursive=True, shared=shared
+            )
+        if 'dims=rz' in self.spec:
+            return find_libraries(
+                ['libwarpx.rz'], root=self.prefix, recursive=True, shared=shared
+            )
```

Comment:
We can make this more compact:
```suggestion
        libsuffix = { '2': '2d', '3': '3d', 'rz': 'rz' }
        dims = self.spec.variants['dims'].value
        return find_libraries(
            ['libwarpx.' + libsuffix[dims]], root=self.prefix, recursive=True,
            shared=True
        )
```

Reply:
@ax3l Thank you for suggesting a fix.

I pushed the fix.